phase2/kubeadm/do

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+set -x
+
+cd "${BASH_SOURCE%/*}"
+
+CONFIG_FILE="${CONFIG_JSON_FILE:-../../.config.json}"
+KUBEADM_INIT_PARAM_FILE=/etc/kubeadm/kubeadm_init_params.txt
+
+upgrade_master() {
+  KUBERNETES_VERSION=$(jq -r '.phase2.kubernetes_version' ${CONFIG_FILE})
+  UPGRADE_METHOD=$(jq -r '.phase2.kubeadm.master_upgrade.method' ${CONFIG_FILE})
+  MASTER=$(do_phase1 master-host)
+
+  update_host_kubeadm ${MASTER}
+  wait_nodes_ready
+
+  INIT_PARAM=$(execute_host ${MASTER} "sudo cat ${KUBEADM_INIT_PARAM_FILE}")
+  case "${UPGRADE_METHOD}" in
+    "init")
+      execute_host ${MASTER} "sudo kubeadm init --skip-preflight-checks --kubernetes-version ${KUBERNETES_VERSION} ${INIT_PARAM}"
+      ;;
+    "upgrade")
+      execute_host ${MASTER} "sudo kubeadm config upload from-flags ${INIT_PARAM}"
+      execute_host ${MASTER} "sudo kubeadm upgrade apply ${KUBERNETES_VERSION} -y -f"
+      ;;
+  esac
+}
+
+wait_nodes_ready() {
+  for TRIES in {1..60}; do
+    NODE_COUNT=$(kubectl get node --no-headers | wc -l)
+    READY_NODE_COUNT=$(kubectl get node --no-headers | grep -c '\sReady\s')
+    if [[ $NODE_COUNT -eq $READY_NODE_COUNT ]]; then
+      echo "All ${NODE_COUNT} nodes ready."
+      return
+    else
+      echo "${READY_NODE_COUNT}/${NODE_COUNT} nodes ready. Waiting for all nodes to become ready... ${TRIES}/60"
+      sleep 5
+    fi
+  done
+  echo "Gave up waiting for nodes to become ready." >&2
+  exit 1
+}
+
+update_host_kubeadm() {
+  HOST=$1 
+  KUBEADM_VERSION=$(jq -r '.phase2.kubeadm.version' ${CONFIG_FILE})
```

Comment:
I know this was done in another PR, but the inconsistency of these two is pretty striking when seen like this. It would be great to get them both using the dot or underscore.

`.phase2.kubeadm.version`
`.phase2.kubelet_version`

Reply:
Will need to think about this and the right approach going forward. I followed the pattern of kubernetes_version since the kubelet is a required part of a cluster just like the control plane is.

Reply:
Fine for now